### PR TITLE
[EOSF-615] Remove maintenance banner from preprints, in ember-osf now

### PR DESCRIPTION
**Pull request by hmoco**
## Ticket

https://openscience.atlassian.net/browse/EOSF-615
## Purpose

[EOSF-615] Allow use of different types of maintenances and custom messages

### CODE REVIEW

**Comment by hmoco** [comment]
@jamescdavis, could you take a look at this? (and do a functional test please?)

File: app/components/maintenance-banner/component.js [hunk 1]
```diff
@@ -1,12 +1,18 @@
 import Ember from 'ember';
 import config from 'ember-get-config';
 
+var levelMap = {
```
Comment (jamescdavis) [inline app/components/maintenance-banner/component.js]: const?

File: app/components/maintenance-banner/component.js [hunk 2]
```diff
@@ -15,6 +21,11 @@ export default Ember.Component.extend({
 
     hidden: 'true',
 
+    alert: true,
+    _info: true,
```
Comment (jamescdavis) [inline app/components/maintenance-banner/component.js]: Why not use a single bound property that's a string? The you can just do something like:
```
classNameBindings: ['alertClass'],

alertClass: 'alert-info',
.
.
.
this.set('alertClass', `alert-${levelMap[maintenance.level]}`);
```

Comment (jamescdavis) [reply app/components/maintenance-banner/component.js]: Also, you can use classNames and classNameBindings simultaneously so you can put `alert` in classNames without needing a bound property.

File: app/components/maintenance-banner/component.js [hunk 3]
```diff
@@ -1,12 +1,18 @@
...
 
+var levelMap = {
+    1: 'info',
+    2: 'warning',
+    3: 'danger'
+};
+
 export default Ember.Component.extend({
-    classNames: [
+    classNamesBindings: [
```
Comment (jamescdavis) [inline app/components/maintenance-banner/component.js]: Should be `classNameBindings` (you have an extra s)

File: app/components/maintenance-banner/component.js [hunk 4]
```diff
@@ -34,6 +45,8 @@ export default Ember.Component.extend({
                             hidden: false,
                             maintenance
                         });
+                        this.set('_info', false);
+                        this.set(`_${levelMap[maintenance.level]}`, true);
```
Comment (jamescdavis) [inline app/components/maintenance-banner/component.js]: Should gracefully handle the case when maintenance.level is `null`, `undefined`, or not mapped in levelMap.

**Comment by jamescdavis** [comment]
Changes look good. Tested and it works! Will merge after tests pass.

**Pull request merged**